dpctl/dparray.py

```diff
@@ -0,0 +1,257 @@
+import numpy as np
+from inspect import getmembers, isfunction, isclass, isbuiltin
+from numbers import Number
+from types import FunctionType as ftype, BuiltinFunctionType as bftype
+import sys
+import inspect
+import dpctl
+from dpctl.memory import MemoryUSMShared
+
+debug = False
+
+
+def dprint(*args):
+    if debug:
+        print(*args)
+        sys.stdout.flush()
+
+
+functions_list = [o[0] for o in getmembers(np) if isfunction(o[1]) or isbuiltin(o[1])]
+class_list = [o for o in getmembers(np) if isclass(o[1])]
+
+array_interface_property = "__array_interface__"
```

Comment:
This really needs to be `__sycl_usm_array_interface__`.

Reply:
Done.